CMakeLists.txt

```diff
@@ -223,31 +231,59 @@ if(NOT NETWORKIT_EXT_TLX)
 else()
 	add_library(tlx STATIC IMPORTED)
 	set_target_properties(tlx PROPERTIES
-			IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/libtlx.a"
+			IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}tlx${CMAKE_STATIC_LIBRARY_SUFFIX}"
 			INTERFACE_INCLUDE_DIRECTORIES "${NETWORKIT_EXT_TLX}/include/")
 endif()
 
 ################################################################################
 # NETWORKIT MODULES
 if(NETWORKIT_BUILD_CORE AND NETWORKIT_MONOLITH)
+
 	if(NETWORKIT_STATIC)
 		add_library(networkit networkit/cpp/networkit.cpp)
 	else()
+		if(WIN32)
+			message(FATAL_ERROR
+					"Windows does only support static builds. Call cmake with '-DNETWORKIT_STATIC=ON'.")
+		endif()
 		add_library(networkit SHARED networkit/cpp/networkit.cpp)
 	endif()
 
+	if(WIN32 AND NETWORKIT_BUILDING_STATELIB)
```

Comment:
What happens if `NETWORKIT_BUILDING_STATELIB` is disabled? Does the build still work? AFAICS it won't because GlobalState.cpp is not compiled in.

Reply:
I would be fine with discarding the configuration option to avoid this problem for now (but of course, we need to keep the `/DNETWORKIT_BUILDING_STATELIB` flag).